src/shadow.rs

```diff
@@ -0,0 +1,174 @@
+//! Manipulate the contents of the shadow password file, `/etc/shadow`.
+use std::ffi::CStr;
+use std::ffi::CString;
+
+/// Represents an entry in `/etc/shadow`.
+// Documentation is based on the `shadow(5)` and `shadow(3)` man pages.
+#[derive(Debug, Clone, Eq, PartialEq, Hash)]
+pub struct Shadow {
+    /// User name
+    pub name: String,
+    /// Encrypted password
+    ///
+    /// Refer to crypt(3) for details on how this string is interpreted.
+    /// If the password field contains some string that is not a valid
+    /// result of crypt(3), for instance ! or *, the user will not be able to
+    /// use a unix password to log in (but the user may log in the system by
+    /// other means).
+    ///
+    /// This field may be empty, in which case no passwords are required to
+    /// authenticate as the specified login name. However, some applications
+    /// which read the /etc/shadow file may decide not to permit any access at
+    /// all if the password field is empty.
+    ///
+    /// A password field which starts with a exclamation mark means that the
+    /// password is locked.
+    pub password: CString,
+    /// Date of the last password change
+    ///
+    /// It is expressed as the number of days since Jan 1, 1970. The value 0
+    /// has a special meaning, which is that the user should change their
+    /// password the next time they will log in the system
+    pub last_change: libc::c_long,
+    /// Minimum password age
+    ///
+    /// The minimum password age is the number of days the user will have to
+    /// wait before she will be allowed to change their password again.
+    /// An empty field and value 0 mean that there are no minimum password age.
+    pub min: libc::c_long,
```

Comment:
Instead of mysterious flag values, how about making this field a `NonZeroIsize`?

Reply:
Just to clarify, do you mean Option<NonZeroIsize>, where zero corresponds to `None`?